quiche/src/lib.rs

```diff
@@ -8306,7 +8426,26 @@ impl TransportParams {
             TransportParams::encode_param(&mut b, 0x000c, 0)?;
         }
 
-        // TODO: encode preferred_address
+        if is_server {
+            if let Some(preferred_address_params) = &tp.preferred_address_params {
+                /// IPv4: 4 bytes
+                /// Port: 2 bytes
+                /// IPv6: 16 bytes
+                /// Port: 2 bytes
+                /// ConnectionID.len(): 1 byte
+                /// ConnectionID: up to 20 bytes
+                /// Stateless Reset Token: 16 bytes
+                const PREFERRED_ADDRESS_PARAM_MAX_SIZE: usize = 61;
+
+                let mut buffer = [0; PREFERRED_ADDRESS_PARAM_MAX_SIZE];
+                let written_to_buffer = PreferredAddressParams::encode(
+                    preferred_address_params,
+                    &mut buffer,
+                )?;
+                TransportParams::encode_param(&mut b, 0x000d, written_to_buffer)?;
+                b.put_bytes(&buffer[0..written_to_buffer])?;
```

Comment:
I guess we have to alloc anyway, so maybe just do that inside PreferredAddressParams::encode and return the buffer there? - Lucas